test/tests/TimerTest.cpp

```diff
@@ -0,0 +1,244 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+#include "gtest/gtest.h"
+#include "Timer.h"
+#include "ServiceMock.h"
+#include "IarmBusMock.h"
+#include "FactoriesImplementation.h"
+
+using namespace WPEFramework;
+
+namespace {
+const string iarmName = _T("Thunder_Plugins");
+}
+
+class TimerTestFixture : public ::testing::Test {
+protected:
+    Core::ProxyType<Plugin::Timer> plugin;
+    Core::JSONRPC::Handler& handler;
+    Core::JSONRPC::Connection connection;
+    string response;
+    ServiceMock service;
+    Core::JSONRPC::Message message;
+    IarmBusImplMock iarmBusImplMock;
+    FactoriesImplementation factoriesImplementation;
+
+    TimerTestFixture()
+        : plugin(Core::ProxyType<Plugin::Timer>::Create())
+        , handler(*(plugin))
+        , connection(1, 0)
+    {
+        PluginHost::IFactories::Assign(&factoriesImplementation);
+    }
+    virtual ~TimerTestFixture()
+    {
+        PluginHost::IFactories::Assign(nullptr);
+    }
+
+    virtual void SetUp()
+    {
+        IarmBus::getInstance().impl = &iarmBusImplMock;
+    }
+
+    virtual void TearDown()
+    {
+        IarmBus::getInstance().impl = nullptr;
+    }
+};
+
+TEST_F(TimerTestFixture, registeredMethods)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("startTimer")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("cancel")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("suspend")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("resume")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getTimerStatus")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getTimers")));
+}
+
+TEST_F(TimerTestFixture, paramsMissing)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("startTimer"), _T("{}"), response));
+    EXPECT_EQ(response, _T("{\"success\":false}"));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("cancel"), _T("{}"), response));
+    EXPECT_EQ(response, _T("{\"success\":false}"));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("suspend"), _T("{}"), response));
+    EXPECT_EQ(response, _T("{\"success\":false}"));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("resume"), _T("{}"), response));
+    EXPECT_EQ(response, _T("{\"success\":false}"));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getTimerStatus"), _T("{}"), response));
+    EXPECT_EQ(response, _T("{\"success\":false}"));
+}
+
+TEST_F(TimerTestFixture, jsonRpc)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("startTimer"), _T("{\"interval\":10}"), response));
+    //EXPECT_THAT(response, testing::MatchesRegex("timerId*"));
+    EXPECT_THAT(response, testing::HasSubstr("timerId"));
```

Comment:
Can you please check more data or ideally match the whole response like you did for "Submit"?